<b><h1>Задание 1</h1></b>

In [1]:
ids = {'user1': [213, 213, 213, 15, 213],
'user2': [54, 54, 119, 119, 119],
'user3': [213, 98, 98, 35]}

geo = set()

for key, item in ids.items():
    geo.update(item)

print(geo)

{98, 35, 15, 213, 54, 119}


Ожидаемый результат: {98, 35, 15, 213, 54, 119}

<b><h1>Задание 2</h1></b>

In [2]:
queries = [
'смотреть сериалы онлайн',
'новости спорта',
'афиша кино',
'курс доллара',
'сериалы этим летом',
'курс по питону',
'сериалы про спорт',
]

num_queries = len(queries)
two_words = 0
three_words = 0

for item in queries:
    num_query_words = len(item.split(' '))
    if num_query_words == 2:
        two_words += 1
    elif num_query_words == 3:
        three_words += 1

print('Поисковых запросов, содержащих 2 слов(а): ' + str(round(two_words/num_queries*100, 2)) + '%')
print('Поисковых запросов, содержащих 3 слов(а): ' + str(round(three_words/num_queries*100, 2)) + '%')

        

Поисковых запросов, содержащих 2 слов(а): 42.86%
Поисковых запросов, содержащих 3 слов(а): 57.14%


Ожидаемый результат:
Поисковых запросов, содержащих 2 слов(а): 42.86%
Поисковых запросов, содержащих 3 слов(а): 57.14%